### PR TITLE
Strategy pattern incorporation

### PR DESCRIPTION
**Pull request by aukgit**
# Title

### CODE REVIEW

File: 09. cli-and-http-request/clihelper/Structs/CmfRunningInfo.go [hunk 1]
```diff
@@ -3,4 +3,5 @@ package clihelper
 type CmdRunningInfo struct {
 	Title, Description                  string
 	ID, ParentId, SessionId, BufferSize int
+	isAsync                             bool
```
Comment (aukgit) [inline 09. cli-and-http-request/clihelper/Structs/CmfRunningInfo.go:6]: `isAsync` -> `IsAsync`

Comment (aukgit) [reply 09. cli-and-http-request/clihelper/Structs/CmfRunningInfo.go:6]: all public members should be CameCase

File: 09. cli-and-http-request/clihelper/WriteUsingErrorLoggers.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package clihelper
+
+func WriteUsingErrortLoggers(cliBindingProperties CliBindingProperties, line string) {
+	_, err := w.Write(line)
```
Comment (aukgit) [inline 09. cli-and-http-request/clihelper/WriteUsingErrorLoggers.go:4]: where is `w` defined?? we should omit this line

File: 09. cli-and-http-request/clihelper/WriteUsingErrorLoggers.go [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+	if err != nil {
```
Comment (aukgit) [inline 09. cli-and-http-request/clihelper/WriteUsingErrorLoggers.go:5]: this if is not necessary

File: 09. cli-and-http-request/clihelper/WriteUsingOutputLoggers.go [hunk 4]
```diff
@@ -1,8 +1,10 @@
 package clihelper
 
-func WriteUsingOutputLoggers(c CliBindingProperties, d []byte) {
-	_, err := w.Writer(c.WritersCollection.WriterConfiguration.WriteToFilelocation, d)
+func WriteUsingOutputLoggers(cliBindingProperties CliBindingProperties, line string) {
+	_, err := w.Writer(line)
```
Comment (aukgit) [inline 09. cli-and-http-request/clihelper/WriteUsingOutputLoggers.go:4]: Same goes here, where is `w` coming from?

File: 09. cli-and-http-request/clihelper/WriteUsingOutputLoggers.go [hunk 5]
```diff
@@ -1,8 +1,10 @@
...
 	if err != nil {
```
Comment (aukgit) [inline 09. cli-and-http-request/clihelper/WriteUsingOutputLoggers.go:5]: if should be omitted

File: 09. cli-and-http-request/clihelper/NewStdInParameter.go [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package clihelper
+
+func NewStdInParameters(cmd *Cmd) *StdInParameter {
```
Comment (aukgit) [inline 09. cli-and-http-request/clihelper/NewStdInParameter.go:3]:
```
func NewStdInParameters(cmd *Cmd) *StdInParameter {	
        outIn, _ = cmd.StdoutPipe()
	errIn, _ = cmd.StderrPipe()
	return *StdInParameter{outIn, errIn}
}
```

File: 09. cli-and-http-request/clihelper/Interfaces/Add.go [hunk 7]
```diff
@@ -0,0 +1,5 @@
+package clihelper
+
+type Add interface {
+	Add(array []Writer)
```
Comment (aukgit) [inline 09. cli-and-http-request/clihelper/Interfaces/Add.go:4]: the single writer should be added

**Review by aukgit — COMMENTED** [review]
Let's apply these fixes @Nafrose

File: 09. cli-and-http-request/clihelper/Interfaces/Add.go (hunk 7 above)
Comment (Nafrose) [reply 09. cli-and-http-request/clihelper/Interfaces/Add.go:4]: done

File: 09. cli-and-http-request/clihelper/NewStdInParameter.go (hunk 6 above)
Comment (Nafrose) [reply 09. cli-and-http-request/clihelper/NewStdInParameter.go:3]: done

File: 09. cli-and-http-request/clihelper/Structs/CmfRunningInfo.go (hunk 1 above)
Comment (Nafrose) [reply 09. cli-and-http-request/clihelper/Structs/CmfRunningInfo.go:6]: Done

File: 09. cli-and-http-request/clihelper/WriteUsingErrorLoggers.go (hunk 2 above)
Comment (Nafrose) [reply 09. cli-and-http-request/clihelper/WriteUsingErrorLoggers.go:4]: w not part of this. sorry

File: 09. cli-and-http-request/clihelper/WriteUsingOutputLoggers.go (hunk 4 above)
Comment (Nafrose) [reply 09. cli-and-http-request/clihelper/WriteUsingOutputLoggers.go:4]: corected

File: 09. cli-and-http-request/clihelper/Structs/StdInParameters.go [hunk 8]
```diff
@@ -4,15 +4,7 @@ type CmdRunner struct {
 	CliBindingProperties CliBindingProperties
 }
 
-type Writer interface {
-	Writer(wc WriterConfiguration, line string)
-}
-
 type StdInParameter struct {
 	stdoutIn []byte
```
Comment (aukgit) [inline 09. cli-and-http-request/clihelper/Structs/StdInParameters.go:8]: `stdoutIn` -> `StdoutIn`

**Pull request closed without merging**